app/api/api/views.py

```diff
@@ -413,6 +422,113 @@ def perform_destroy(self, instance):
                 raise Exception(f"Error deleting data dictionary: {e}")
         instance.delete()
 
+    def create(self, request, *args, **kwargs):
+        non_file_serializer = ScanReportCreateSerializerNonFiles(
+            data=request.data, context={"request": request}
+        )
+        if not non_file_serializer.is_valid():
+            return Response(
+                non_file_serializer.errors, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        file_serializer = self.get_serializer(data=request.FILES)
+        if not file_serializer.is_valid():
+            return Response(file_serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        self.perform_create(file_serializer, non_file_serializer)
+        headers = self.get_success_headers(file_serializer.data)
+        return Response(
+            file_serializer.data, status=status.HTTP_201_CREATED, headers=headers
+        )
+
+    def perform_create(self, serializer, non_file_serializer):
+        validatedFiles = serializer.validated_data
+        validatedData = non_file_serializer.validated_data
+        # List all the validated data and files
+        valid_data_dictionary_file = validatedFiles.get("data_dictionary_file")
+        valid_scan_report_file = validatedFiles.get("scan_report_file")
+        valid_visibility = validatedData.get("visibility")
+        valid_editors = validatedData.get("editors")
+        valid_dataset = validatedData.get("dataset")
+        valid_parent_dataset = validatedData.get("parent_dataset")
+
+        rand = "".join(random.choices(string.ascii_lowercase + string.digits, k=8))
+        dt = "{:%Y%m%d-%H%M%S}".format(datetime.datetime.now())
+        print(dt, rand)
+        # Create an entry in ScanReport for the uploaded Scan Report
+        scan_report = ScanReport.objects.create(
+            dataset=valid_dataset,
+            parent_dataset=valid_parent_dataset,
+            name=modify_filename(valid_scan_report_file, dt, rand),
+            visibility=valid_visibility,
+        )
+
+        scan_report.author = self.request.user
+        scan_report.save()
+
+        # # Add viewers to the scan report if specified
+        # if sr_viewers := request.data.get("viewers"):
+        #     scan_report.viewers.add(*sr_viewers)
+
+        # Add editors to the scan report if specified
+        if sr_editors := valid_editors:
+            scan_report.editors.add(*sr_editors)
+
+        # Grab Azure storage credentials
+        blob_service_client = BlobServiceClient.from_connection_string(
+            os.getenv("STORAGE_CONN_STRING")
+        )
+
+        print("FILE >>> ", str(valid_scan_report_file))
+        print("STRING TEST >>>> ", scan_report.name)
+
+        # If there's no data dictionary supplied, only upload the scan report
+        # Set data_dictionary_blob in Azure message to None
+        if str(valid_data_dictionary_file) == "undefined":
```

Comment:
For all this upload code - the download, and delete are now in this service: https://github.com/Health-Informatics-UoN/Carrot-Mapper/blob/master/app/api/mapping/services.py
We should move this over also (maybe a future PR)